deploy/crds/devopsconsole_v1alpha1_gitsource_crd.yaml

```diff
@@ -0,0 +1,41 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: gitsources.devopsconsole.openshift.io
+spec:
+  group: devopsconsole.openshift.io
+  names:
+    kind: GitSource
+    listKind: GitSourceList
+    plural: gitsources
+    singular: gitsource
+    shortNames:
+      - gitsrc
+  scope: Namespaced
+  validation:
+    # openAPIV3Schema is the schema for validating custom objects.
+    openAPIV3Schema:
+      properties:
+        apiVersion:
+          type: string
+        kind:
+          type: string
+        metadata:
+          type: object
+        spec:
+          properties:
+            url:
+              type: string
+            ref:
+              type: string
+            contextDir:
+              type: string
```

Comment:
I hope this is meant to be the directory which we intend to use as the source directory. Consider a repo with frontend and backend all in one. If  the GitSource just wants to build the backend, I guess the `contextDir` will be set to `backend`, correct?

Do you want the `contextDir` to be respected by the build tool dectector so that it only looks inside of it and not in e.g. the `frontend` directory?

Reply:
yes, it's just a sub folder. Could be especially useful for monorepos.

Reply:
Then we don't need that information here. We should just focus on the git source and let the build tool decide which folder to use. Or at least it should be in the domain of the build tool and not of the source tool IMHO.

Reply:
What if there are multiple apps in the same repo and they should be build independently? A monorepo. Basically speaking why if some git repo contains multiple independent git sources?
IMO we should let users to specify that optional subfolder.

Reply:
What if there are multiple apps in the same repo and they should be build independently? A monorepo. Basically speaking why if some git repo contains multiple independent git sources?
IMO we should let users to specify that optional subfolder.

Reply:
@alexeykazakov if there are multiple apps in a mono-repo that you want to build independently I'd say we should be able to re-use a `GitSource` in two build configurations and have each build configuaration define a dfifferent `contextDir`. @sbose78 thoughts?

Reply:
@kwk 
The buildConfig object would be more of an 'implementation detail' which the controller creates on finding that a new gitSource object has been created ( either as `oc create gitsource` or from the UI ).

So the initial information needs to be captured in the initial object ( gitSource CR ) that gets created.

Reply:
Okay, after some talking to @michaelkleinhenz I think I get that GitSource isn't meant to represent a cloned repo. It is just the information on how to get to the source. So if there's a frontend and a backend that both build from the same repo, they should have different GitSource objects that point to the same git. But since it is the build teams job to pick a directory that shall be built, it should be their responsibility to configure the contextDir. It is not part of a representation of how to access a git source which is what GitSource is doing in the end.

Reply:
> It is not part of a representation of how to access a git source which is what GitSource is doing in the end.

I'm not sure I'm following... 
Let's say I have a repo with a hundred of apps there. Each of them are supposed to built independently. Don't ask me why I decided to organize my repo in that way. I just like it.
Now I want to import my `<githost>/coolapp` golang app. Edit it in Che. Build it and deploy. I do not want to import my other nodejs app `<githost>/coolapp-doc` which is just a legacy documentation app. How build/che etc are supposed to guess what I really want to import/edit/build without me, as a user, explicitly point to `<githost>/coolapp` when importing the app?